src/elements/Input/Input.js

```diff
@@ -88,7 +88,13 @@ function Input(props) {
   )
   const rest = getUnhandledProps(Input, props)
   const ElementType = getElementType(Input, props)
-  const inputProps = _.pick(props, htmlInputPropNames)
+  const inputProps = {
+    ..._.pick(props, htmlInputPropNames),
+    // React gives a warning if no onChange is given to a controlled input. We
+    // handle the onChange on the wrapping element, so we don't actually need
+    // onChange on the input itself but let's pass a noop to suppress the warning.
+    onChange: _.noop,
```

Comment:
Perhaps we instead add `onChange` to the `htmlInputPropNames`?  This way, it is applied to the correct place and suppresses the warning only when it should be suppressed (user passed value/onChange).  My concern is that we're swallowing the warning.  If the user adds `<Input value='' />`, they should see the warning but they won't with the current change here.